### PR TITLE
CL-271 Show a notification when a configuration change is needed for your print

### PR DESCRIPTION
**Pull request by sedwards2009**
https://jira.ultimaker.com:8443/browse/CL-271

When Cura is running, but not as the front-most application, at the time a printer gets blocked because my print is scheduled next and requires a configuration change, I want to receive a proper* desktop notification, so I'm reminded to change the configuration as soon as possible.

### CODE REVIEW

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py [hunk 1]
```diff
@@ -475,6 +476,38 @@ def __filterOurPrintJobs(self, print_jobs):
...
+        old_change_required_print_jobs = self.__filterConfigChangePrintJobs(self.__filterOurPrintJobs(old_print_jobs))
+        new_change_required_print_jobs = self.__filterConfigChangePrintJobs(self.__filterOurPrintJobs(new_print_jobs))
+        old_change_required_print_job_uuids = set([pj["uuid"] for pj in old_change_required_print_jobs])
+
+        for print_job in new_change_required_print_jobs:
+            if print_job["uuid"] not in old_change_required_print_job_uuids:
+
+                printer_name = self.__getPrinterNameFromUuid(print_job["assigned_to"])
+                if printer_name is None:
+                    printer_name = i18n_catalog.i18nc("@info:status", "Unknown printer")
```
Comment (nallath) [inline plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: Is this in the translation catalog already?

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py [hunk 2]
```diff
@@ -475,6 +476,38 @@ def __filterOurPrintJobs(self, print_jobs):
...
+
+                message_text = (i18n_catalog.i18n("{printer_name} is reserved to print '{job_name}'.Please change the printer's configuration to match the job, for it to start printing.")
```
Comment (nallath) [inline plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: Is this in the translation catalog already?

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py [hunk 3]
```diff
@@ -475,6 +476,38 @@ def __filterOurPrintJobs(self, print_jobs):
...
+                                .format(printer_name=printer_name, job_name=print_job["name"]))
+                message = Message(text=message_text, title=i18n_catalog.i18nc("@label:status", "Action required"))
+                Application.getInstance().showMessage(message)
+                Application.getInstance().showToastMessage(
+                    i18n_catalog.i18nc("@label:status", "Action required"),
```
Comment (nallath) [inline plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: Is this in the translation catalog already?

**Comment by nallath** [comment]
Please also provide information what the pull request is about for those of us that don't have acces to Jira.

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py (hunk 1 above)
Comment (jackha) [reply plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: No... I'll try to find a matching string

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py (hunk 2 above)
Comment (jackha) [reply plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: Yes

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py (hunk 3 above)
Comment (jackha) [reply plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: This string is part of CL-506, it's going to be.

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py (hunk 1 above)
Comment (jackha) [reply plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: It's going to be: msgctxt "@label"
msgid "Unknown"

File: plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py [hunk 4]
```diff
@@ -475,6 +476,38 @@ def __filterOurPrintJobs(self, print_jobs):
...
+
+        for print_job in new_change_required_print_jobs:
+            if print_job["uuid"] not in old_change_required_print_job_uuids:
+
+                printer_name = self.__getPrinterNameFromUuid(print_job["assigned_to"])
+                if printer_name is None:
+                    printer_name = i18n_catalog.i18nc("@info:status", "Unknown printer")
+
+                message_text = (i18n_catalog.i18n("{printer_name} is reserved to print '{job_name}'. Please change the printer's configuration to match the job, for it to start printing.")
+                                .format(printer_name=printer_name, job_name=print_job["name"]))
```
Comment (nallath) [inline plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: () around the catalog isn't required.

Comment (jackha) [reply plugins/UM3NetworkPrinting/NetworkClusterPrinterOutputDevice.py]: Well it is, unless you put "\" at the end of the line

**Pull request merged**